ui/packages/shared/parser/src/index.ts

```diff
@@ -45,6 +45,54 @@ function matcherTypeFromString(matcherTypeString: string): MatcherType {
   }
 }
 
+function findMatcherLabelName(stateStack: any): string {
+  let currentState = stateStack.find((e: any) => e.rule.name === 'matcher');
+
+  if (currentState === undefined) {
+    return '';
+  }
+
+  while (currentState.right.rule.name !== 'labelName') {
+    currentState = currentState.left;
+  }
+
+  return currentState.right.data.value;
+}
+
+export interface LiteralSuggestion {
+  type: 'literal';
```

Comment:
let's maybe save these type scripts to a Typescript enum? Looks like some of the values are re-used so can help us avoid misspellings. 

For example,
```
export enum SuggestionType {
  Literal = 'literal',
  Matcher = 'matcherType',
  LabelName = 'labelName',
  LabelValue = 'labelValue',
  ProfileName = 'profileName'
}
```

Reply:
they're not reused they're used to select the right concrete type from the `Suggestion` union type, because of the union type, they already act as as enum at compile time

Reply:
Ok, gotcha! Just wanted to make sure a potentially misspelled suggestion type string would be flagged as a ts error, if needed

Reply:
FYI TS `enum` is to be avoided, it is not fully type safe, there are better alternatives, please see #1633 for details